tools/cleanlab/cleanlab_issue_handler.py

```diff
@@ -0,0 +1,154 @@
+import argparse
+
+import numpy as np
+import pandas as pd
+from xgboost import XGBClassifier
+
+from cleanlab import Datalab
+from cleanlab.regression.rank import get_label_quality_scores
+from sklearn.linear_model import LinearRegression
+from sklearn.model_selection import KFold, StratifiedKFold, cross_val_predict
+from sklearn.neighbors import NearestNeighbors
+
+
+# -------------------
+# Issue Handler
+# -------------------
+class IssueHandler:
+
+    def __init__(self, dataset, task, n_splits=3, quality_threshold=0.2, knn_k=10):
+        self.dataset = dataset
+        self.task = task
+        self.n_splits = n_splits
+        self.quality_threshold = quality_threshold
+        self.knn_k = knn_k
+        self.issues = None
+        self.features = None
+        self.knn_graph = None
+        self.features = self.dataset.drop('target', axis=1).columns.tolist()
+        self.pred_probs = None
+        self.issue_summary = None
+
+    def report_issues(self):
+        X = self.dataset.drop('target', axis=1)
+        y = self.dataset['target']
+
+        # Ensure compatibility with Galaxy!
+        X = X.to_numpy() if hasattr(X, 'to_numpy') else np.asarray(X)
+        y = y.to_numpy() if hasattr(y, 'to_numpy') else np.asarray(y)
+
+        # Compute knn_graph
+        nn = NearestNeighbors(n_neighbors=self.knn_k + 1)
+        nn.fit(X)
+        self.knn_graph = nn.kneighbors(return_distance=False)[:, 1:]
+
+        if self.task == 'classification':
+            model = XGBClassifier(use_label_encoder=False, eval_metric='logloss', random_state=42)
+            cv = StratifiedKFold(n_splits=self.n_splits, shuffle=True, random_state=42)
+            self.pred_probs = cross_val_predict(model, X, y, cv=cv, method='predict_proba')
```

Comment:
`self.pred_probs` is used only for the classification method. Why not use just a local variable (without `self`)?